projects/plugins/jetpack/class.jetpack.php

```diff
@@ -868,6 +863,11 @@ public function configure() {
 			$this->connection_manager = new Connection_Manager( 'jetpack' );
 		}
 
+		$modules = new Automattic\Jetpack\Modules();
+		if ( $modules->is_active( 'publicize' ) && $this->connection_manager->has_connected_user() ) {
+			$config->ensure( 'publicize' );
+		}
+
```

Comment:
I don't know how much I like doing this check here. It would be better that the module wasn't active if there was not connection, or that we handle the check for the connection in the Publicize package setup code.

Could we prevent the package from being `ensure`d if some requirements aren't met?

Maybe this is ok for now.

Reply:
Maybe we can do something similar to the check that happens with offline mode: https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/class.jetpack-admin.php#L392

If the site is in Offline Mode and the module requires a (user) connection, [the module gets bypassed](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/class.jetpack.php#L1860). I wonder if we can do something similar early on in our init code.

Reply:
I've been playing with things a bit, and I think we might need a better way of conditionally ensuring fetaures in Config. I'll see if there's a better way, but we might want to merge this first so that it can be included in the Jetpack release